biogrid/libs/biogrid-simulator/src/lib/biogrid/biogrid-class.ts

```diff
@@ -144,15 +151,27 @@ export class Biogrid implements Grid {
   // TODO pass a list of equal length to hold the area for the solar panels
   private createSolarPanels(positions: ItemPosition[]): EnergySource[] {
     return positions.map(
-      (position, index) => new SolarPanel({
-        x: position.x,
-        y: position.y,
-        efficiency: 0.75,
-        areaSquareMeters: SOLAR_PANEL.AREA,
-        gridItemName: `${GRID_ITEM_NAMES.SOLAR_PANEL}-${index}`
-      } as SolarPanelParams)
+      (position, index) =>
+        new SolarPanel({
+          x: position.x,
+          y: position.y,
+          efficiency: 0.75,
+          areaSquareMeters: SOLAR_PANEL.AREA,
+          gridItemName: `${GRID_ITEM_NAMES.SOLAR_PANEL}-${index}`,
+          date: this.startDate,
+        } as SolarPanelParams)
     );
   }
+
+  /**
+   * Drain the energy users according to the time of day
+   */
+  drainEnergyUsers(date: Date) {
+    this.town.getEnergyUsers().forEach((energyUser) => {
+      energyUser.decreaseEnergyAccordingToTimeOfDay(date);
+    });
```

Comment:
I think this naming is pretty confusing. Energy users are things like houses right? So you're not really "draining" them of energy, but rather providing them with energy that they need. Can we rename?

Reply:
Oh it's more like updating energy usage. Changed name to `updateEnergyUsage`